src/peripherals/usart.c

```diff
@@ -221,8 +228,16 @@ void usarts_enable(u32 ftdi_baud, u32 uarta_baud, u32 uartb_baud, bool do_precon
   chBSemInit(&uartb_state.claimed, FALSE);
   uartb_state.configured = true;
 
-  all_uarts_enabled = true;
+  if (do_configure_hooks) {
```

Comment:
Its not immediately obvious why these hooks aren't together with the radio ones. Maybe unify them so they are using the claim mechanism in the same way? or at least add a comment saying why they are separated.